node/src/archivist.rs

```diff
@@ -0,0 +1,91 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use crate::database::archive::SQLiteArchive;
+use crate::database::Archivist;
+use crate::{database, vm, LongLivedService, Network};
+use async_trait::async_trait;
+use node_data::archive::ArchivalData;
+use std::sync::Arc;
+use tokio::sync::mpsc::Receiver;
+use tracing::error;
+
+pub struct ArchivistSrv {
+    pub archive_receiver: Receiver<ArchivalData>,
+    pub archivist: SQLiteArchive,
+}
+
+#[async_trait]
+impl<N: Network, DB: database::DB, VM: vm::VMExecution>
+    LongLivedService<N, DB, VM> for ArchivistSrv
+{
+    async fn execute(
+        &mut self,
+        _: Arc<tokio::sync::RwLock<N>>,
```

Comment:
can do `use tokio::snyc::RwLock` (you already have a tokio `use` statement above you can extend it also) and just use `RwLock<N>`